src/apply.jl

```diff
@@ -101,3 +101,27 @@ function apply!(table::T, t::Transform; cols=_get_cols(table), kwargs...)::T whe
 
     return table
 end
+
+"""
+    apply_append(A::AbstractArray, ::Transform; append_dim, kwargs...)
+
+Applies the [`Transform`](@ref) to `A` and returns the result in a new array where the output
+is appended to `A` along the `append_dim` dimension. The remaining `kwargs` correspond to
+the usual [`Transform`](@ref) being invoked.
+"""
+function apply_append(A::AbstractArray, t; append_dim, kwargs...)::AbstractArray
+    return cat(A, apply(A, t; kwargs...); dims=append_dim)
```

Comment:
I guess this copies the axiskeys of the chunk being applied to, so there isn't the equivalent of `header`?

I can understand not needing this, given axis keys are more like metadata for each data point. For example even if you take `Power(3)` of the data, it still corresponds to `time=t`.

What's your thinking on it?

Reply:
>I guess this copies the axiskeys of the chunk being applied to, so there isn't the equivalent of header?

Correct, I went looking in the AxisKeys code for something like it but [`new_keys` is set internally](https://github.com/mcabbott/AxisKeys.jl/blob/master/src/functions.jl#L172) and can't be over-written externally.

>What's your thinking on it?

For transforms that affect the whole (2D) Array, it makes sense to append on the 3rd dimension and create a new dim that way. Otherwise you end up with redundant keys in one dimension.

For linear combination, or transforms on single columns it seems unavoidable for now that you have to rekey on the other side. We could do that processing here? But it would require loading AxisKeys explicitly to allow new keys be provided. For now I'm happy to leave it and see how much of an issue it becomes.

Reply:
I think I'll leave this for another PR - I have an idea or two of how to make this work but it'll require another review